### PR TITLE
KOGITO-2181 DMN decision service single output decision coercion

### PR DESCRIPTION
**Pull request by tarilabs**

### CODE REVIEW

File: kogito-codegen/src/main/resources/class-templates/DMNRestResourceTemplate.java [hunk 1]
```diff
@@ -33,7 +40,31 @@ public Object dmn(java.util.Map<String, Object> variables) {
...
+    
+    private static final com.fasterxml.jackson.databind.ObjectMapper objectMapper = new com.fasterxml.jackson.databind.ObjectMapper()
+            .registerModule(new com.fasterxml.jackson.datatype.jsr310.JavaTimeModule())
+            .registerModule(new com.fasterxml.jackson.databind.module.SimpleModule()
+                            .addSerializer(org.kie.dmn.feel.lang.types.impl.ComparablePeriod.class,
+                                           new org.kie.kogito.dmn.rest.DMNFEELComparablePeriodSerializer()))
+            .disable(com.fasterxml.jackson.databind.SerializationFeature.WRITE_DATES_AS_TIMESTAMPS)
+            .disable(com.fasterxml.jackson.databind.SerializationFeature.WRITE_DURATIONS_AS_TIMESTAMPS);
+
+    private Object extractSingletonDSIfSucceded(DMNResult result) {
```
Comment (hellowdan) [inline kogito-codegen/src/main/resources/class-templates/DMNRestResourceTemplate.java:61]: not sure if this method is in use

Comment (tarilabs) [reply kogito-codegen/src/main/resources/class-templates/DMNRestResourceTemplate.java:61]: yes I confirm it is once codegenerated

File: integration-tests/pom.xml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  <parent>
+    <groupId>org.kie.kogito</groupId>
+    <artifactId>kogito-runtimes</artifactId>
+    <version>8.0.0-SNAPSHOT</version>
+  </parent>
+  <artifactId>integration-tests</artifactId>
+  <packaging>pom</packaging>
+
+  <modules>
+    <module>integration-tests-quarkus</module>
```
Comment (evacchi) [inline integration-tests/pom.xml:12]: does this mean we'll have to change the archetype on Quarkus platform? @mariofusco

Comment (evacchi) [reply integration-tests/pom.xml:12]: I think that relying on the quarkus extension testing frameworks should be good enough for these: `kogito-quarkus-extension/integration-test`, so you should be able move them there.

**Comment by tarilabs** [comment]
```

[INFO] ------------------------------------------------------------------------

[INFO] Reactor Summary for Kogito Runtimes 8.0.0-SNAPSHOT:

[INFO] 

[INFO] Kogito Runtimes .................................... SUCCESS [04:33 min]

[INFO] Kogito BOM ......................................... SUCCESS [  0.018 s]

[INFO] Kogito :: API Parent ............................... SUCCESS [  0.017 s]

[INFO] Kogito :: Public API ............................... SUCCESS [01:33 min]

[INFO] Kogito :: Internal ................................. SUCCESS [  2.416 s]

[INFO] Kogito :: Services ................................. SUCCESS [  6.780 s]

[INFO] Drools Multiproject ................................ SUCCESS [  0.019 s]

[INFO] Drools :: Core :: Reflective ....................... SUCCESS [  0.208 s]

[INFO] Drools :: Core :: Static ........................... SUCCESS [  0.126 s]

[INFO] Drools :: Core :: Dynamic .......................... SUCCESS [  1.801 s]

[INFO] Drools :: Core ..................................... SUCCESS [01:01 min]

[INFO] Drools :: Compiler ................................. SUCCESS [03:16 min]

[INFO] kogito-drools-model ................................ SUCCESS [  3.742 s]

[INFO] kogito-ruleunits ................................... SUCCESS [  2.583 s]

[INFO] kogito-drools ...................................... SUCCESS [  0.042 s]

[INFO] Kogito :: Decision Model Notation .................. SUCCESS [ 34.924 s]

[INFO] drools-decisiontables .............................. SUCCESS [ 14.431 s]

[INFO] Kogito :: Scenario Simulation Runner ............... SUCCESS [  7.631 s]

[INFO] jBPM :: Multiproject ............................... SUCCESS [  0.011 s]

[INFO] jBPM :: Flow ....................................... SUCCESS [  8.895 s]

[INFO] jBPM :: Flow Builder ............................... SUCCESS [ 34.296 s]

[INFO] jBPM :: BPMN2 ...................................... SUCCESS [02:19 min]

[INFO] jBPM :: Serverless Workflow ........................ SUCCESS [ 54.710 s]

[INFO] Kogito Cloud Services .............................. SUCCESS [  0.015 s]

[INFO] Kogito Cloud Kubernetes Client ..................... SUCCESS [01:01 min]

[INFO] Kogito Cloud Work Items ............................ SUCCESS [  8.058 s]

[INFO] Kogito OptaPlanner integration (wrapper) ........... SUCCESS [ 17.671 s]

[INFO] grafana-api ........................................ SUCCESS [  2.766 s]

[INFO] kogito-codegen ..................................... SUCCESS [01:54 min]

[INFO] kogito-springboot .................................. SUCCESS [01:32 min]

[INFO] Kogito Springboot Starter .......................... SUCCESS [  0.042 s]

[INFO] Add-Ons to the runtimes ............................ SUCCESS [  0.011 s]

[INFO] Prometheus Monitoring AddOn ........................ SUCCESS [  5.675 s]

[INFO] Kogito Persistence AddOns .......................... SUCCESS [  0.012 s]

[INFO] Kogito - Infinispan persistence .................... FAILURE [01:06 min]

[INFO] infinispan-quarkus-health-addon .................... FAILURE [02:13 min]

[INFO] Kogito - File System persistence ................... SUCCESS [  3.710 s]

[INFO] Kogito Events ...................................... SUCCESS [  0.013 s]

[INFO] Kogito Events Reactive Messaging AddOn ............. SUCCESS [ 32.542 s]

[INFO] Kogito Events SprintBoot AddOn ..................... SUCCESS [  8.656 s]

[INFO] Jobs ............................................... SUCCESS [  0.014 s]

[INFO] Jobs API ........................................... SUCCESS [  1.728 s]

[INFO] jobs-management-common ............................. SUCCESS [  1.780 s]

[INFO] Jobs Management Quarkus Addon ...................... SUCCESS [ 35.655 s]

[INFO] Jobs Management SprintBoot Addon ................... SUCCESS [ 43.065 s]

[INFO] Kogito Predictions ................................. SUCCESS [  0.014 s]

[INFO] Kogito Predictions API ............................. SUCCESS [  2.709 s]

[INFO] Kogito Predictions SMILE addon ..................... SUCCESS [  7.093 s]

[INFO] process-management ................................. SUCCESS [  0.016 s]

[INFO] process-management-common .......................... SUCCESS [  7.888 s]

[INFO] Process Management Addon for Quarkus ............... SUCCESS [  3.772 s]

[INFO] process-management-springboot-addon ................ SUCCESS [  3.833 s]

[INFO] tracing ............................................ SUCCESS [  0.012 s]

[INFO] kogito-tracing-decision-common ..................... SUCCESS [  9.621 s]

[INFO] kogito-tracing-decision-quarkus-addon .............. SUCCESS [ 36.008 s]

[INFO] Kogito :: Maven Plugin ............................. SUCCESS [01:46 min]

[INFO] Kogito - Quarkus Extension ......................... SUCCESS [  0.014 s]

[INFO] Kogito - Quarkus - Extension BOM ................... SUCCESS [  0.012 s]

[INFO] Kogito - Quarkus - Runtime ......................... SUCCESS [ 39.643 s]

[INFO] Kogito - Quarkus - Deployment ...................... SUCCESS [01:26 min]

[INFO] Kogito - Quarkus - Integration Tests ............... SUCCESS [ 47.204 s]

[INFO] Kogito - Quarkus - Integration Tests (Hot Reload) .. SUCCESS [ 39.353 s]

[INFO] Kogito - Quarkus - Integration Tests (Legacy API) .. SUCCESS [ 21.135 s]

[INFO] Kogito Maven Archetypes ............................ SUCCESS [  0.013 s]

[INFO] SpringBoot Kogito Archetype ........................ SUCCESS [01:06 min]

[INFO] Quarkus Kogito Archetype ........................... SUCCESS [ 27.857 s]

[INFO] integration-tests .................................. SUCCESS [  0.014 s]

[INFO] integration-tests-quarkus .......................... SUCCESS [ 20.702 s]

[INFO] integration-tests-springboot ....................... SUCCESS [01:24 min]

[INFO] ------------------------------------------------------------------------

[INFO] BUILD FAILURE

[INFO] ------------------------------------------------------------------------

[INFO] Total time:  36:55 min

[INFO] Finished at: 2020-05-19T11:04:42-04:00

[INFO] ------------------------------------------------------------------------

[ERROR] Failed to execute goal org.apache.maven.plugins:maven-failsafe-plugin:2.22.1:verify (default) on project infinispan-persistence-addon: There are test failures.

[ERROR] 

[ERROR] Please refer to /home/jenkins/workspace/IE_kogito_kogito-runtimes_PR-515/addons/persistence/infinispan-persistence-addon/target/failsafe-reports for the individual test results.

[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.

[ERROR] -> [Help 1]

[ERROR] Failed to execute goal org.apache.maven.plugins:maven-failsafe-plugin:2.22.1:verify (default) on project infinispan-quarkus-health-addon: There are test failures.

[ERROR] 

[ERROR] Please refer to /home/jenkins/workspace/IE_kogito_kogito-runtimes_PR-515/addons/persistence/infinispan-quarkus-health-addon/target/failsafe-reports for the individual test results.
```

File: integration-tests/integration-tests-springboot/src/it/integration-tests-springboot-it/pom.xml [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+      <groupId>org.springframework.boot</groupId>
+      <artifactId>spring-boot-starter-test</artifactId>
+      <scope>test</scope>
+    </dependency>
+    <!-- Manually ovverride to 3.3.0 because Spring boot uses rest-assured-common@3.3.0
+         Otherwise error: The type io.restassured.common.mapper.TypeRef cannot be resolved. It is indirectly referenced from required .class files -->
+    <dependency>
+      <groupId>io.rest-assured</groupId>
+      <artifactId>rest-assured</artifactId>
+      <version>3.3.0</version>
```
[the remark this answers is not in the document]
Comment (tarilabs) [reply integration-tests/integration-tests-springboot/src/it/integration-tests-springboot-it/pom.xml]: done

File: integration-tests/integration-tests-springboot/src/it/integration-tests-springboot-it/src/test/java/org/kie/kogito/integrationtests/springboot/DSCoercionTest.java [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+    @Test
+    public void testDSdtd() {
+        Duration DSdtd = given().contentType(ContentType.JSON)
+                                .when()
+                                .post("/DScoercion/DSdtd")
+                                .getBody().as(Duration.class);
+        assertThat(DSdtd, is(Duration.parse("PT1H")));
+    }
+
+    // DROOLS-5344
```
[the remark this answers is not in the document]
Comment (tarilabs) [reply integration-tests/integration-tests-springboot/src/it/integration-tests-springboot-it/src/test/java/org/kie/kogito/integrationtests/springboot/DSCoercionTest.java]: done

**Comment by tarilabs** [comment]
for gatekeepers: Edo contacted me earlier asking for more clarification and we need to sync on this, please hold any merge.

**Comment by tarilabs** [comment]
discussed online with Edo, since https://github.com/kiegroup/kogito-runtimes/pull/515#pullrequestreview-415065303 removing the "do not merge" label

**Pull request merged**